File: problems/258_add-digits/__main__.py
class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10:
            return num
        sum = 0
        while num:
            sum += num % 10
            num //= 10
        return self.addDigits(sum)


print(Solution().addDigits(38))

File: problems/258_add-digits/main.go
package main

import (
	"fmt"
	"testing"
)

func addDigits(num int) int {
	if num < 10 {
		return num
	}
	var sum int
	for num > 0 {
		sum += num % 10
		num /= 10
	}
	return addDigits(sum)
}

func Test258(t *testing.T) {
	fmt.Println(addDigits(38))
	t.Error()
}
